gschem/include/prototype.h

```diff
@@ -234,7 +234,7 @@ void i_action_update_status(GschemToplevel *w_current,gboolean inside_action);
 void i_show_state(GschemToplevel *w_current, const char *message);
 void i_set_state(GschemToplevel *w_current, enum x_states newstate);
 void i_set_state_msg(GschemToplevel *w_current, enum x_states newstate, const char *message);
-void i_update_middle_button(GschemToplevel *w_current, void (*func_ptr)(), const char *string);
+void i_update_middle_button (GschemToplevel *w_current, void (*func_ptr) (gpointer, guint, GtkWidget*), const char *string);
```

Comment:
You have written out this function type many times in this patch series!  Maybe it should have a `typedef`?